examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -142,20 +142,31 @@ class _PestoDemoState extends State<PestoDemo> {
           new DrawerItem(
             child: new Text('Home'),
             selected: !config.showFavorites,
-            onPressed: () => Navigator.pushNamed(context, PestoDemo.routeName)
+            onPressed: () {
+              Navigator.popUntil(context, (Route<dynamic> route) {
+                return !route.willHandlePopInternally
+                  && route is ModalRoute && route.settings.name == '/pesto';
```

Comment:
I'd thought about providing a utility function ModalRoute.withName(route, name) but it didn't seem useful enough.  A ModalRoute.withName(name) that returns a predicate (and factors in willHandlePopInternally, which would be the common case)  would be pretty slick.